include/zephyr/sensing/sensing_sensor.h

```diff
@@ -153,20 +204,47 @@ extern struct sensing_sensor SENSING_SENSOR_SOURCE_NAME(idx, node);	\
 				(,), node, cb_list_ptr)			\
 	};
 
+/**
+ * @brief Structure for sensor submit configuration.
+ *
+ * This structure represents a sensor submit configuration. It includes the channel, info index, and
+ * streaming flag.
+ */
 struct sensing_submit_config {
-	enum sensor_channel chan;
-	const int info_index;
-	const bool is_streaming;
+	enum sensor_channel chan; /**< Channel of the sensor submit configuration. */
+	const int info_index;     /**< Info index of the sensor submit configuration. */
+	const bool is_streaming;  /**< Streaming flag of the sensor submit configuration. */
 };
 
+/**
+ * @brief External declaration for the sensing I/O device API.
+ *
+ * This external declaration represents the sensing I/O device API.
+ */
```

Comment:
see earlier comment re: internal. Do not hesitate to put internal stuff inside `@cond INTERNAL_HIDDEN` blocks. You may be more concise with the documentation for these (and don't even need to use doxygen syntax) as they're not meant for external consumption

Reply:
Ok, thanks.